bin/mympirun.py

```diff
@@ -40,83 +40,139 @@
 @author: Stijn De Weirdt, Jens Timmerman (HPC UGent / VSC)
 """
 
-import sys
+import glob
+import inspect
 import os
+import sys
+import traceback
 
-from vsc.utils import fancylogger
-from vsc.mympirun.mpi.factory import getinstance
-from vsc.mympirun.mpi.mpi import whatMPI
+import vsc.mympirun.mpi.mpi
 from vsc.mympirun.option import MympirunOption
-from vsc.mympirun.rm.sched import whatSched
+from vsc.mympirun.rm.factory import getinstance
+import vsc.mympirun.rm.sched
+from vsc.utils import fancylogger
 
 _logger = fancylogger.getLogger()
-
-
-class ExitException(Exception):
-    """Exception thrown when we wish to exit, but no real errors occured"""
-    pass
-
+fancylogger.setLogLevelDebug()
 
 def get_mpi_and_sched_and_options():
-    """Parses the mpi and scheduler based on current environment and guesses the best one to use"""
-    scriptname, mpi, found_mpi = whatMPI(sys.argv[0])
-
+    """selects mpi flavor and scheduler based on environment and arguments
+
+    returns:
+    mpi             -- the chosen MPI flavor
+    sched           -- the chosen scheduler
+    mo              -- the MympirunOption class
+    """
+    _logger.debug("mympirun.py - get_mpi_and_sched_and_options()")
+
+    # import the various MPI flavours and Schedulers
+    import_implementations(vsc.mympirun.mpi.mpi)
+    import_implementations(vsc.mympirun.rm.sched)
+
+    # see if an mpi flaver was explicitly chosen as an argument
+    # if not, just use the mpirun that was called
+    # We are using sys.argv because generaloption depends on the the returned
+    # scriptname
+    setmpi_index = sys.argv.index("-M") if "-M" in sys.argv else 0
```

Comment:
okay, I hadn't thought about this. We should then just use a env variable. This meddling with `sys.argv` is too error prone. Check for `MYMPIRUN_FLAVOUR`?